File: src/test/java/io/github/zot201/asmhook/test/HookTest.java
package io.github.zot201.asmhook.test;

import com.google.testing.compile.JavaFileObjects;
import io.github.zot201.asmhook.processing.HookProcessor;
import org.junit.Test;

import static com.google.common.truth.Truth.assert_;
import static com.google.testing.compile.JavaSourceSubjectFactory.javaSource;

public class HookTest {

  @Test public void helloWorld() {
    assert_().about(javaSource())
        .that(JavaFileObjects.forResource("HelloWorld.java"))
        .processedWith(new HookProcessor())
        .compilesWithoutError();
  }

}

File: processor/src/test/scala/io/github/zot201/asmhook/test/HookTest.scala
package io.github.zot201.asmhook.test

import com.google.testing.compile.JavaFileObjects
import io.github.zot201.asmhook.processing.HookProcessor
import io.github.zot201.asmhook.test.util.Truths._
import org.junit.Test

import scala.collection.JavaConversions._

class HookTest {

  def dummy(file: String) = JavaFileObjects.forResource(s"dummy/$file.java")
  def root(file: String) = JavaFileObjects.forResource(s"$file.java")

  val itemRes = Set(dummy("Item"))
  val enumEnchantmentTypeRes = itemRes + dummy("EnumEnchantmentType")
  val helloWorldRes = enumEnchantmentTypeRes + root("HelloWorld")

  @Test def helloWorld(): Unit = {
    assertAboutJavaSources
      .that(helloWorldRes)
      .processedWith(new HookProcessor)
      .compilesWithoutError()
  }

}
